backend/app/models/budget.rb

```diff
@@ -0,0 +1,103 @@
+# budgets table
+class Budget < ActiveRecord::Base
+
+  validates :name,        presence: true
+  validates :description, presence: true
+  validates :contact,     presence: true
+  validates :email,       presence: true
+  validates :phone,       presence: true
+
+  # Return all budgets.
+  #
+  # @return all budgets
+  def self.find_all
+    Budget.order(:name)
+  end
+
+  # Return all budgets beginning with fitst letter l ('*' as non-alphanumeric wildcard).
+  #
+  # @return all budgets beginning with fitst letter l ('*' as non-alphanumeric wildcard)
+  def self.find_letter(l)
```

Comment:
this name suggests you are trying to find the letter, maybe `find_by_first_letter` ?

Reply:
Done.